### PR TITLE
[wptserve] Make .headers work with generated tests

### PR DESCRIPTION
**Pull request by Hexcles**
Honour *.headers (including *.sub.headers and __dir__{.sub}.headers)
when serving tests automatically generated from *.js files. The
documentation doesn't mention this exception and reasonable people would
assume *.headers work the same way as with static files. Add a
functional test, too.

Fixes https://crbug.com/1135559.

Drive-by: some stylistic changes and docstrings.

### CODE REVIEW

File: tools/wptserve/wptserve/handlers.py [hunk 1]
```diff
@@ -161,6 +167,36 @@ def wrap_pipeline(path, request, response):
     return response
 
 
+def load_headers(request, path):
+    """Loads headers from files for a given path.
+
+    Attempt to load both the neighbouring __dir__{.sub}.headers and
```
Comment (stephenmcgruer) [inline tools/wptserve/wptserve/handlers.py:173]:
```suggestion
    Attempts to load both the neighbouring __dir__{.sub}.headers and
```

File: tools/wptserve/wptserve/handlers.py [hunk 2]
```diff
@@ -161,6 +167,36 @@ def wrap_pipeline(path, request, response):
...
+    PATH{.sub}.headers (apply template substituition if needed); results are
```
Comment (stephenmcgruer) [inline tools/wptserve/wptserve/handlers.py:174]:
```suggestion
    PATH{.sub}.headers (applying template substitution if needed); results are
```

File: tools/wptserve/wptserve/handlers.py [hunk 3]
```diff
@@ -161,6 +167,36 @@ def wrap_pipeline(path, request, response):
...
+    concatenated in this order.
```
Comment (stephenmcgruer) [inline tools/wptserve/wptserve/handlers.py:175]:
```suggestion
    concatenated in that order.
```

File: tools/serve/serve.py [hunk 4]
```diff
@@ -73,7 +73,9 @@ def __call__(self, request, response):
         self.handler(request, response)
 
     def handle_request(self, request, response):
-        for header_name, header_value in self.headers:
+        headers = self.headers + handlers.load_headers(
```
Comment (stephenmcgruer) [inline tools/serve/serve.py:76]: Optional: I'd consider renaming `self.headers` to something like `self.base_headers`. It's just a bit weird to say "these are the headers, but then we have to add more headers on-top to make up the headers".

**Review by stephenmcgruer — APPROVED** [review]

File: tools/serve/serve.py (hunk 4 above)
Comment (Hexcles) [reply tools/serve/serve.py:76]: This is pretty much an implementation detail (despite being public), as it's not intended/suitable for direct use, so I'm not very worried about the name here.

**Pull request merged**